frontend/drivers/platform_ctr.c

```diff
@@ -1,564 +1,584 @@
+   fill_pathname_join(g_defaults.path.config, g_defaults.dirs[DEFAULT_DIR_PORT],
+                      file_path_str(FILE_PATH_MAIN_CONFIG), sizeof(g_defaults.path.config));
+}
+
+static void frontend_ctr_deinit(void* data)
+{
+   Handle lcd_handle;
+   u32 parallax_layer_reg_state;
+   u8 not_2DS;
+
+   extern PrintConsole* currentConsole;
+
+   (void)data;
+
+#ifndef IS_SALAMANDER
+   verbosity_enable();
+
+#ifdef HAVE_FILE_LOGGER
+   command_event(CMD_EVENT_LOG_FILE_DEINIT, NULL);
+#endif
+
+   if ((gfxBottomFramebuffers[0] == (u8*)currentConsole->frameBuffer)
+         && (ctr_fork_mode == FRONTEND_FORK_NONE))
+      wait_for_input();
+
+   CFGU_GetModelNintendo2DS(&not_2DS);
+
+   if (not_2DS && srvGetServiceHandle(&lcd_handle, "gsp::Lcd") >= 0)
+   {
+      u32* cmdbuf = getThreadCommandBuffer();
+      cmdbuf[0]   = 0x00110040;
+      cmdbuf[1]   = 2;
+      svcSendSyncRequest(lcd_handle);
+      svcCloseHandle(lcd_handle);
+   }
+
+   parallax_layer_reg_state = (*(float*)0x1FF81080 == 0.0) ? 0x0 : 0x00010001;
+   GSPGPU_WriteHWRegs(0x202000, &parallax_layer_reg_state, 4);
+
+   mcuHwcExit();
+   ptmuExit();
+   cfguExit();
+   ndspExit();
+   csndExit();
+   gfxTopRightFramebuffers[0] = NULL;
+   gfxTopRightFramebuffers[1] = NULL;
+   gfxExit();
+#endif
+}
+
+static void frontend_ctr_exec(const char* path, bool should_load_game)
+{
+   int error = 0;
+   char args[0x300 - 0x4];
+
+   DEBUG_VAR(path);
+   DEBUG_STR(path);
+
+   args[0] = '\0';
+   strcat(args, "\"");
+   strcat(args, elf_path_cst);
+   strcat(args, "\"");
+
+   RARCH_LOG("Attempt to load core: [%s].\n", path);
+#ifndef IS_SALAMANDER
+   if (should_load_game && !path_is_empty(RARCH_PATH_CONTENT))
+   {
+	  strcat(args, " \"");
+      strcat(args, path_get(RARCH_PATH_CONTENT));
+	  strcat(args, "\"");
+      RARCH_LOG("content path: [%s].\n", path_get(RARCH_PATH_CONTENT));
+   }
+#endif
+
+   if (path && path[0])
+   {
+#ifdef IS_SALAMANDER
+      struct stat sbuff;
+      bool file_exists;
+
+      file_exists = stat(path, &sbuff) == 0;
+      if (!file_exists)
+      {
+         char core_path[PATH_MAX];
+
+         /* find first valid core and load it if the target core doesnt exist */
+         get_first_valid_core(&core_path[0]);
+
+         if (core_path[0] == '\0')
+         {
+            errorConf error;
+
+            errorInit(&error, ERROR_TEXT, CFG_LANGUAGE_EN);
+            errorText(&error, "There are no cores installed, install a core to continue.");
+            errorDisp(&error);
+            exit(0);
+         }
+      }
+#endif
+      if (envIsHomebrew())
+      {
+         exec_3dsx_no_path_in_args(path, args);
+      }
+      else
+      {
+         RARCH_LOG("\n");
+         RARCH_LOG("\n");
+         RARCH_LOG("Warning:\n");
+         RARCH_LOG("First core launch may take 20 seconds!\n");
+         RARCH_LOG("Do not force quit before then or your memory card may be corrupted!\n");
+         RARCH_LOG("\n");

```

Comment:
maybe add a progress display during cia install ?
you could also add a prompt for `install`/`cancel` but that is not necessary.